docs/current/guides/snippets/use-services/start-stop-services/index.ts

```diff
@@ -0,0 +1,46 @@
+import { connect, Client } from "@dagger.io/dagger"
+
+connect(
+  async (client: Client) => {
+    const dockerd = await client
+      .container()
+      .from("docker:dind")
+      .asService()
+      .start()
+
+    // dockerd is now running, and will stay running
+    // so you don't have to worry about it restarting after a 10 second gap
+
+    // then in all of your tests, continue to use an explicit binding:
+    const test = await client
+      .container()
+      .from("golang")
+      .withServiceBinding("docker", dockerd)
+      .withEnvVariable("DOCKER_HOST", "tcp://docker:2375")
+      .withExec(["go", "test", "./..."])
```

Comment:
For that execution, I'm getting the following output:

```shell
21: exec go test ./...
21: [0.06s] pattern ./...: directory prefix . does not contain main module or its selected dependencies
21: exec go test ./... ERROR: process "go test ./..." did not complete successfully: exit code: 1

9: start dockerd-entrypoint.sh
9: > in service idpdkck56djam
```

Reply:
Renamed `main.go` to `main_test.go` which should fix it, but I haven't been able to figure out how to test this locally.